mythril/laser/ethereum/instructions.py

```diff
@@ -448,11 +413,15 @@ def calldatacopy_(self, global_state):
                 return [global_state]
 
             try:
-                i_data = environment.calldata[dstart]
+                i_data = dstart
 
+                new_memory = []
                 for i in range(mstart, mstart + size):
-                    state.memory[i] = environment.calldata[i_data]
-                    i_data += 1
+                    new_memory.append(environment.calldata[i_data])
+                    i_data = simplify(i_data + 1)
```

Comment:
this will break if i_data is an int right?

Reply:
Or can that never happen?

Reply:
Great catch. I think it might be able to happen.

Reply:
Good catch. I think it can happen.